nion/swift/model/Graphics.py

```diff
@@ -1019,19 +1021,28 @@ def _rotated_bottom_left(self) -> Geometry.FloatPoint:  # useful for testing
         return rotate(self._bounds.bottom_left, self._bounds.center, self.rotation)
 
     def get_mask(self, data_shape: DataAndMetadata.ShapeType, calibrated_origin: typing.Optional[Geometry.FloatPoint] = None) -> DataAndMetadata._ImageDataType:
-        mask = numpy.zeros(data_shape)
-        bounds_int = ((int(data_shape[0] * self.bounds[0][0]), int(data_shape[1] * self.bounds[0][1])),
-                      (int(data_shape[0] * self.bounds[1][0]), int(data_shape[1] * self.bounds[1][1])))
         if self.rotation:
-            a, b = bounds_int[0][0] + bounds_int[1][0] * 0.5, bounds_int[0][1] + bounds_int[1][1] * 0.5
-            y, x = numpy.ogrid[-a:data_shape[0] - a, -b:data_shape[1] - b]  # type: ignore
-            angle_sin = math.sin(self.rotation)
-            angle_cos = math.cos(self.rotation)
-            mask_eq = (numpy.fabs(x * angle_cos - y * angle_sin) / (bounds_int[1][1] / 2) <= 1) & (numpy.fabs(y * angle_cos + x * angle_sin) / (bounds_int[1][0] / 2) <= 1)  # type: ignore
-            mask[mask_eq] = 1
+            bounds = Geometry.FloatRect.make(self.bounds)
+            mask = make_rectangle_mask(data_shape=data_shape, center=bounds.center, size=bounds.size, rotation=self.rotation)
         else:
-            mask[bounds_int[0][0]:bounds_int[0][0] + bounds_int[1][0] + 1,
-                 bounds_int[0][1]:bounds_int[0][1] + bounds_int[1][1] + 1] = 1
+            mask = numpy.zeros(data_shape)
```

Comment:
This section can be eliminated. Call `make_rectangle_mask` in all cases. Put the `rotation` = 0 optimization into `make_rectangle_mask`. See the comment in `make_rectangle_mask`.